src/ConfirmSelectEventPublisher.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/*
+ * This file is part of the prooph/humus-amqp-producer.
+ * (c) 2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare (strict_types=1);
+
+namespace Prooph\ServiceBus\Message\HumusAmqp;
+
+use Humus\Amqp\Producer;
+use Prooph\Common\Event\ActionEvent;
+use Prooph\EventStore\EventStore;
+use Prooph\EventStore\Plugin\Plugin;
+use Prooph\ServiceBus\EventBus;
+use Prooph\ServiceBus\Exception\RuntimeException;
+
+/**
+ * Class ConfirmSelectEventPublisher
+ * @package Prooph\ServiceBus\Message\HumusAmqp
+ */
+final class ConfirmSelectEventPublisher implements Plugin
+{
+    /**
+     * @var EventBus
+     */
+    private $eventBus;
+
+    /**
+     * @var Producer
+     */
+    private $producer;
+
+    /**
+     * @var float
+     */
+    private $timeout;
+
+    /**
+     * @var ActionEvent[]
+     */
+    private $queuedActionEvents = [];
+
+    /**
+     * @var bool
+     */
+    private $inConfirmSelectMode = false;
+
+    /**
+     * ConfirmSelectEventPublisher constructor.
+     * @param EventBus $eventBus
+     * @param Producer $producer
+     * @param float $timeout
+     */
+    public function __construct(EventBus $eventBus, Producer $producer, float $timeout = 2.0)
```

Comment:
amqp allows fractions of seconds, like 0.1 or 1.5, 2.0 is only a reasonable default.